book/crud.rst

```diff
@@ -0,0 +1,477 @@
+
+    // src/InventoryBundle/Form/Type/VehicleType.php
+    namespace InventoryBundle\Form\Type;
+
+    use Symfony\Component\Form\AbstractType;
+    use Symfony\Component\Form\FormBuilderInterface;
+    use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+    class VehicleType extends AbstractType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('model')
+                ->add('seats')
+                ->add('boughtAt')
+                ->add('leasedUntil')
+            ;
+        }
+
+        public function setDefaultOptions(OptionsResolverInterface $resolver)
+        {
+            $resolver->setDefaults(array(
+                'data_class' => 'InventoryBundle\Entity\Vehicle',
+            ));
+        }
+
+        public function getName()
+        {
+            return 'inventory.vehicle';
+        }
+    }
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/InventoryBundle/Resources/config/form.yml
+    services:
+        inventory.form.type.vehicle:
+            class: InventoryBundle\Form\Type\VehicleType
+            tags:
+                - { name: form.type, alias: inventory.vehicle }
+
+Then, you will need to create the needed controller actions. You can simplify the actions if you
+let the ``createAction`` submit the form to the ``updateAction``. Therefore, you have to make sure
+that the ``updateAction`` accepts requests without an existing ``Vehicle`` object being passed:
+
+.. code-block:: php
+    :linenos:
+
+    // src/InventoryBundle/Controller/VehicleController.php
+    namespace InventoryBundle\Controller;
+
+    // ...
+    use InventoryBundle\Entity\Vehicle;
+
+    class VehicleController extends Controller
+    {
+        /**
+         * @Route("/create", name="inventory.vehicle_create")
+         * @Template("InventoryBundle:Vehicle:update.html.twig")
+         * @Acl(
+         *     id="inventory.vehicle_create",
+         *     type="entity",
+         *     class="InventoryBundle:Vehicle",
+         *     permission="CREATE"
+         * )
+         */
+        public function createAction()
+        {
+            return array(
+                'entity' => new Vehicle(),
+                'form' => $this->get('inventory.form.type.vehicle')->createView(),
+            );
+        }
+
+        /**
+         * @Route("/update/{id}", name="inventory.vehicle_update", requirements={"id":"\d+"}, defaults={"id":0})
+         * @Template
+         * @Acl(
+         *     id="inventory.vehicle_update",
+         *     type="entity",
+         *     class="InventoryBundle:Vehicle",
+         *     permission="EDIT"
+         * )
+         */
+        public function updateAction(Vehicle $vehicle = null)
+        {
+            $vehicle = $vehicle ?: new Vehicle();
+
+            return array(
+                'entity' => $vehicle,
+                'form' => $this->get('inventory.form.type.vehicle')->createView(),
```

Comment:
We can't use "createView()" on the form type. We need to define a form that's using this form type:
```yaml
    inventory.vehicle.form:
        class: Symfony\Component\Form\Form
        factory_method: createNamed
        factory_service: form.factory
        arguments:
            - ~
            - "inventory_vehicle"
```